htlcswitch/switch.go

```diff
@@ -621,9 +710,18 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, attemptID uint64,
 			false,
 		)
 
-		return errFeeExposureExceeded
+		return nil, errFeeExposureExceeded
 	}
 
+	return link, nil
+}
+
+// commitHTLCCircuit creates and commits a payment circuit for the given HTLC.
+// It handles duplicate detection and attaches the circuit to the packet if
+// successful.
+func (s *Switch) commitHTLCCircuit(packet *htlcPacket,
```

Comment:
<img width="937" height="556" alt="Screenshot 2025-12-05 at 22 38 21" src="https://github.com/user-attachments/assets/fc3c1991-c388-4a42-9a1a-6b23246bbbab" />


can we test the duplicate cases here in a unit test.